### DISCUSSION TITLE
Object type for props support

### DISCUSSION THREAD

**Discussion by demo88** [zauberzeug/nicegui#706] [Ideas / Feature Requests]
PROPS_PATTERN in element.py may not support an object type, such as dict, list. I confirmed dict was not supported when I tried to use the locale props of [QDate](https://quasar.dev/vue-components/date#qdate-api) .

I updated PROPS_PATTERN to add "{}" and "[]" pattern, and modified the check logic in the "_parse_props" method.

Wish a better solution for this problem and release soon  ^o^

BTW, this is an amazing project, THX !!!

**Comment by falkoschindler** [comment]
Glad to hear that you like the project!

It is already possible to pass dictionaries and lists to `.props()`, as shown here: https://github.com/zauberzeug/nicegui/issues/457#issuecomment-1477351513.

But supporting such objects without quotes would be nice. Would you mind sharing your solution and maybe some demo code? Or even better, would you like to provide a pull request? 🙂

**Comment by falkoschindler** [comment]
This feature has been implemented in PR #4969 which is released in version 2.23.
